src/inference/inference.py

```diff
@@ -1,37 +1,48 @@
-import torch
 import os
-from peft import AutoPeftModelForCausalLM
-from transformers import AutoTokenizer, pipeline
-import gradio as gr
+from typing import List
+
+import gradio as gr  # type: ignore
```

Comment:
I see this `type: ignore` comment in a couple different places. Could you explain the reasoning @Siddhesh-Agarwal

Reply:
This is used by `mypy` to ignore the type of the variable (sometimes due to incompatible datatypes). I'll remove it if it is unnecessary for this project.

Reply:
I don't believe it is needed for this